datadog/resource_datadog_synthetics_global_variable.go

```diff
@@ -0,0 +1,143 @@
+package datadog
+
+import (
+	"strings"
+
+	datadogV1 "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceDatadogSyntheticsGlobalVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogSyntheticsGlobalVariableCreate,
+		Read:   resourceDatadogSyntheticsGlobalVariableRead,
+		Update: resourceDatadogSyntheticsGlobalVariableUpdate,
+		Delete: resourceDatadogSyntheticsGlobalVariableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tags": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceDatadogSyntheticsGlobalVariableCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV1 := providerConf.DatadogClientV1
+	authV1 := providerConf.AuthV1
+
+	syntheticsGlobalVariable := buildSyntheticsGlobalVariableStruct(d)
+	createdSyntheticsGlobalVariable, _, err := datadogClientV1.SyntheticsApi.CreateGlobalVariable(authV1).Body(*syntheticsGlobalVariable).Execute()
+	if err != nil {
+		// Note that Id won't be set, so no state will be saved.
+		return translateClientError(err, "error creating synthetics global variable")
+	}
+
+	// If the Create callback returns with or without an error without an ID set using SetId,
+	// the resource is assumed to not be created, and no state is saved.
+	d.SetId(createdSyntheticsGlobalVariable.GetId())
+
+	// Return the read function to ensure the state is reflected in the terraform.state file
+	return resourceDatadogSyntheticsGlobalVariableRead(d, meta)
+}
+
+func resourceDatadogSyntheticsGlobalVariableRead(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV1 := providerConf.DatadogClientV1
+	authV1 := providerConf.AuthV1
+
+	syntheticsGlobalVariable, _, err := datadogClientV1.SyntheticsApi.GetGlobalVariable(authV1, d.Id()).Execute()
+
+	if err != nil {
+		if strings.Contains(err.Error(), "404 Not Found") {
```

Comment:
You should be able to use the HTTP response instead and check the status code. I know we used that pattern elsewhere, but it's super awkward.